integration/plugin/authz/authz_plugin_v2_test.go

```diff
@@ -0,0 +1,178 @@
+// +build !windows
+
+package authz
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/integration/internal/api/container"
+	"github.com/docker/docker/integration/internal/api/plugin"
+	"github.com/docker/docker/integration/internal/api/volume"
+	"github.com/docker/docker/integration/util/requirement"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	authzPluginName            = "riyaz/authz-no-volume-plugin"
+	authzPluginTag             = "latest"
+	authzPluginNameWithTag     = authzPluginName + ":" + authzPluginTag
+	authzPluginBadManifestName = "riyaz/authz-plugin-bad-manifest"
+	nonexistentAuthzPluginName = "riyaz/nonexistent-authz-plugin"
+)
+
+func setupTestV2(t *testing.T) func() {
+	skip.IfCondition(t, testEnv.DaemonInfo.OSType != "linux")
+	skip.IfCondition(t, !requirement.HasHubConnectivity())
+
+	teardown := setupTest(t)
+
+	d.Start(t)
+
+	return teardown
+}
+
+func TestAuthZPluginV2AllowNonVolumeRequest(t *testing.T) {
+	skip.IfCondition(t, os.Getenv("DOCKER_ENGINE_GOARCH") != "amd64")
+	defer setupTestV2(t)()
+
+	client, err := d.NewClient()
+	require.Nil(t, err)
+
+	// Install authz plugin
+	err = plugin.InstallGrantAllPermissions(client, authzPluginNameWithTag)
+	require.Nil(t, err)
+	// start the daemon with the plugin and load busybox, --net=none build fails otherwise
+	// because it needs to pull busybox
+	d.Restart(t, "--authorization-plugin="+authzPluginNameWithTag)
+	d.LoadBusybox(t)
+
+	// defer disabling the plugin
+	defer func() {
+		d.Restart(t)
```

Comment:
If we are restarting the daemon on every test anyway why bother leaving it running and having to do extra cleanup?  Why not just stop it and recreate the daemon?

That should allow us to remove all the `d.Start()` at the end of test, which is really unexpected.

Reply:
I believe the daemon is restarted in order to disable the authz plugin so that it does not interfere with cleanup. It may not be necessary but the original tests do this.

Reply:
A big reason for creating this new test suite is to fix the numerous problems in the old suite. 

There are actually two daemons involved here. First there is a single daemon started before any of the test suites are run. Let's call this the "global daemon". Most tests hit this shared daemon, which is why Protect/Clean are required, so the next test can be run against a clean daemon.

These auth tests (also most swarm tests, or anything that requires specific daemon flags) start a new daemon within the test or test suite. Let's call this the "inner daemon".

There shouldn't be any reason to stop, restart, or clean up the inner daemon. We just stop it, and remove the root directory to wipe out any state. All these extra `Restart()` just add to confusion about the suite.

Could we try removing all of these `Restart()`calls and just do a proper clean by stopping and removing the temporary directory?